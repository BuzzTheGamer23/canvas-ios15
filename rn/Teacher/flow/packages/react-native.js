// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
