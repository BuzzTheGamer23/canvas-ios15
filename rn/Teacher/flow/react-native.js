//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
