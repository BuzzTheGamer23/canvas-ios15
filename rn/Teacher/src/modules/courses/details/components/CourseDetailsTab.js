//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
* Launching pad for navigation for a single course
* @flow
*/

import React from 'react'
import {
} from 'react-native'

import Images from '../../../../images'
import Row from '../../../../common/components/rows/Row'

type Props = {
  tab: Tab,
  courseColor: string,
  onPress: Function,
  attendanceTabID: ?string,
}

export default class CourseDetails extends React.Component<any, Props, any> {

  onPress = () => {
    const tab = this.props.tab
    this.props.onPress(tab)
  }

  render () {
    const tab = this.props.tab
    return (<Row
                title={tab.label}
                image={tab.id === this.props.attendanceTabID ? Images.course.attendance : Images.course[tab.id]}
                imageTint={this.props.courseColor}
                imageSize={{ height: 24, width: 24 }}
                onPress={this.onPress}
                disclosureIndicator={true}
                border={'bottom'}
                testID={`courses-details.${tab.id}-cell`}
                titleStyles={{ marginLeft: -4, fontWeight: '500' }}/>)
  }
}
