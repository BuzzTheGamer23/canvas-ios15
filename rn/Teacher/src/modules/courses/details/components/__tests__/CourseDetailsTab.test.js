/* @flow */

import 'react-native'
import React from 'react'
import CourseDetailsTab from '../CourseDetailsTab.js'

const template = {
  ...require('../../../../../api/canvas-api/__templates__/tab'),
}

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const defaultProps = {
  tab: template.tab(),
  courseColor: 'white',
  onPress: () => {},
}

test('renders correctly', () => {
  let tree = renderer.create(
    <CourseDetailsTab {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('default icon', () => {
  const props = {
    ...defaultProps,
    tab: template.tab({ id: 'test default icon' }),
  }

  let tree = renderer.create(
    <CourseDetailsTab {...props} />
  )

  expect(tree).toMatchSnapshot()
})
