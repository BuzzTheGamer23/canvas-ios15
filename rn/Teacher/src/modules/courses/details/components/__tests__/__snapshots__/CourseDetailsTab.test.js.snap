// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        undefined,
        Object {
          "borderBottomColor": "#C7CDD1",
          "borderBottomWidth": 0.5,
        },
      ],
    ]
  }
  testID="courses-details.test default icon-cell"
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontWeight": "600",
              },
              Object {
                "fontWeight": "500",
                "marginLeft": -4,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderColor": "#c7c7cc",
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "height": 10,
              "marginLeft": 7,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 10,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        undefined,
        Object {
          "borderBottomColor": "#C7CDD1",
          "borderBottomWidth": 0.5,
        },
      ],
    ]
  }
  testID="courses-details.assignments-cell"
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "marginRight": 16,
            "resizeMode": "contain",
          },
          Object {
            "height": 24,
            "tintColor": "white",
            "width": 24,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontWeight": "600",
              },
              Object {
                "fontWeight": "500",
                "marginLeft": -4,
              },
            ],
            Object {
              "fontFamily": ".SFUIDisplay-medium",
            },
          ]
        }
      >
        Home
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderColor": "#c7c7cc",
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "height": 10,
              "marginLeft": 7,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 10,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
